### ISSUE TITLE
AES128 vs AES256

### ISSUE THREAD

**Issue by jhass** [sibiantony/ssleuth#15]
This is more a question than a correction.

You rank AES256 higher than AES128, while [Mozilla's recommended setup](https://wiki.mozilla.org/Security/Server_Side_TLS) prefers AES128 over AES256. Their [reasoning](https://wiki.mozilla.org/Security/Server_Side_TLS#Prioritization_logic) is that AES256 doesn't seem to have actual benefits over AES128 and that AES128 might actually be more resilient to timing attacks. [Interesting part of the linked discussion](http://www.mail-archive.com/dev-tech-crypto@lists.mozilla.org/msg11292.html).

So I'm wondering, why not rank them on the same level?

**Comment by sibiantony** [comment]
2 things.
* The ranking were and will not consider performance as a criteria [ as in the case of AES-128 vs AES-256 - something I had discussed/clarified in the security channel when I started development of this add-on ]
* The key length is a criteria when ranking an algorithm, and as long as computing power increases I believe it matters.

But other than that, I agree with you on many things. 
AES-128 is quite strong enough for today's computing capabilities. Many security conscious service providers stick to AES-128, and to me, 'AES-256' is more of a marketing term.

Ranking AES-128 and AES-256 at the same level was considered before . As far as I understand the attacks on AES-256 is only for specific 'rounds'. There are no known weaknesses for 14-round AES. Also, because of the first 2 reasons, I had kept the ranking different considering key length also as a parameter.

In short - this 128vs256 thing is kind of debatable. I am totally okay to change it if there are more arguments/community request. Thanks for bringing this up. Will keep the issue open for input.

**Comment by nbraud** [comment]
The consensus in the crypto community seems to be that the key-schedule issues in AES-192 and AES-256 are more likely to lead to a break than it is for an attack to reduce AES-128's keyspace enough for an attack to be practical.

**Comment by sibiantony** [comment]
Okay. I will make AES128 and AES256 ratings same in a future release. There might also be other  changes (GCM/CBC etc.). If you wish to propose changes, you can find the current ratings here : https://github.com/sibiantony/ssleuth/blob/master/modules/cipher-suites.js

**Comment by sibiantony** [comment]
Changes released in version 0.4.2. Closing issue.
